### ISSUE TITLE
[BUG] cmal_deterministic._search_quantile contracts root bracket via frac_confine=0.8, clipping negative/tail quantiles

### ISSUE THREAD

**Issue by ashermetzger-rgb** [google-research/flood-forecasting#274]
**Location**
`googlehydrology/utils/cmal_deterministic.py:128-135`

**Root Cause Mechanism**
Clamping bounds are calculated via contraction factors (`0.8 * min(ppfs)` and `0.8 * max(ppfs)`). For negative values—common in standardized hydrologic target data and low-flow percentiles ($q \le 0.1$)—multiplying by $0.8$ shifts the lower bound upward toward zero. Restricting $k$ to this contracted interval prevents Newton-Raphson root-finding from reaching solutions in the true mixture domain $[\min(\text{PPF}), \max(\text{PPF})]$.

**Impact**
Distorts tail quantile forecasts (e.g., $q = 0.1$ low flows and $q = 0.9$ peak flood discharge) generated by the `cmal_deterministic` head by prematurely pinning estimates to artificial boundary limits.

**Suggested Fix**
Replace scalar contraction with an additive outward margin expansion:

```
min_ppf = torch.min(ppfs, dim=2, keepdim=True).values
max_ppf = torch.max(ppfs, dim=2, keepdim=True).values
margin = 0.1 * (max_ppf - min_ppf + 1e-6)

low = min_ppf - margin
high = max_ppf + margin

```

**Verification Checklist**

* [ ] Run unit tests on negative normalized inputs across edge percentiles ($q \in \{0.01, 0.05, 0.1, 0.9, 0.95, 0.99\}$).
* [ ] Verify Newton-Raphson iteration convergence rates on validation catchments without boundary clipping flags.
* [ ] Confirm broadcast tensor shapes along `dim=2` across multi-basin batched inference runs.

**Comment by grey-nearing** [comment]
**Confirming this is real and unfixed on `main`, and it's more severe than the report suggests: on confident predictions the deterministic quantiles come back a flat 20% low.**

I hit this while reviewing unrelated CMAL work and measured it against a bisection ground truth. Reproduced at `main` @ `482a773`.

### Mechanism

[`googlehydrology/utils/cmal_deterministic.py:127-135`](https://github.com/google-research/flood-forecasting/blob/main/googlehydrology/utils/cmal_deterministic.py#L127-L135):

```python
ppfs = _ppf(quantile, mu, b, tau)
low = frac_confine * torch.min(ppfs, dim=2, keepdim=True).values   # frac_confine = 0.8
high = frac_confine * torch.max(ppfs, dim=2, keepdim=True).values

k = torch.mean(ppfs, dim=2, keepdim=True)
for _ in range(iterations):
    cdf_val, pdf_val = _mixture_cdf_and_pdf(k, mu, b, tau, pi)
    k = k - (cdf_val - quantile) / (pdf_val + epsilon)
    k = torch.clamp(k, low, high)
```

The comment on `frac_confine` says *"to avoid overshooting"*, but multiplying both ends by 0.8 scales the bracket **toward zero** rather than shrinking it toward the centre of the component PPFs. The true root is not guaranteed to lie inside `[0.8·min, 0.8·max]`.

The degenerate case is exact. When the mixture components agree — `min(ppfs) == max(ppfs) == v` — the bracket collapses to `[0.8v, 0.8v]`, so the `clamp` pins `k` on the very first iteration and **Newton–Raphson never runs**. The function returns `0.8 × truth` by construction.

### Measurements

Bisection on the repo's own `_mixture_cdf_and_pdf` as ground truth:

| Mixture | Quantile | True | `_search_quantile` | Error | Bracket |
|---|---|---|---|---|---|
| identical kernels (confident) | 0.50 | 10.0000 | 8.0000 | **−20.00%** | `[8.000, 8.000]` ← pinned |
| near-identical kernels | 0.50 | 10.0000 | 8.1600 | **−18.40%** | `[7.840, 8.160]` ← pinned |
| low flow, identical kernels | 0.50 | 0.5000 | 0.4000 | **−20.00%** | `[0.400, 0.400]` ← pinned |
| identical kernels, upper tail | 0.90 | 13.2189 | 10.5751 | **−20.00%** | `[10.575, 10.575]` ← pinned |
| asymmetric `tau`, identical `mu` | 0.50 | 11.5250 | 9.2200 | **−20.00%** | `[9.220, 9.220]` ← pinned |
| well-spread kernels | 0.50 | 9.7601 | 9.7601 | ✅ +0.00% | `[1.600, 20.000]` |

### Why this hasn't been caught

The last row is the reason. `test/test_cmal_math.py:97` uses a well-spread fixture, where `[0.8·min, 0.8·max]` still happens to bracket the root, so Newton–Raphson converges normally and the test passes. The bug only appears as the components converge.

That's an unfortunate blind spot: **the error grows as the model gets more confident**. Agreement among mixture components is normally the signal you'd trust most, and it's precisely the regime that's biased here. For an operational flood model, a systematic 20% underestimate of discharge on confident forecasts is worth prioritising.

### Suggested direction

The bracket should contain the root, then be widened rather than contracted — roughly:

```python
low = torch.min(ppfs, dim=2, keepdim=True).values
high = torch.max(ppfs, dim=2, keepdim=True).values
margin = 0.25 * (high - low).clamp(min=some_floor)
low, high = low - margin, high + margin
```

The `some_floor` term matters: without it the degenerate case still collapses to a zero-width bracket. Any fix should come with a regression test using a **degenerate (identical-kernel) fixture**, since that's the case the current suite never exercises.

Happy to put up that fix and test if it's useful.

<sub>Measured with a standalone bisection-vs-`_search_quantile` probe; independently reproduced twice. Analysis assisted by AI, reviewed by me.</sub>
